### PR TITLE
Support "disable_simple_one_line_block" for specific type of code block.

### PR DESCRIPTION
**Pull request by 5lbBookOfGre**
### Summary
* More granular control over "keep_simple_block_one_line".
* This should be backward compatible. 
* Note, I haven't updated README.md or other docs yet; I will do that once I get approval for this change. 

Thank you!

### CODE REVIEW

**Comment by tammela** [comment]
I would move your tests cases from `test/test_config.cpp` to the same structure we use in `test/test_format_file.cpp`. Other than that the feature is much welcome in my opinion.

Thanks!

File: src/FormatVisitor.h [hunk 1]
```diff
@@ -21,8 +23,7 @@ class FormatVisitor : public LuaBaseVisitor {
     antlrcpp::Any visitVarDecl(LuaParser::VarDeclContext* context) override;
     antlrcpp::Any visitGotoStat(LuaParser::GotoStatContext* context) override;
     antlrcpp::Any visitDoStat(LuaParser::DoStatContext* context) override;
-    antlrcpp::Any visitWhileStat(LuaParser::WhileStatContext* context) override;
-    antlrcpp::Any visitRepeatStat(LuaParser::RepeatStatContext* context) override;
+    antlrcpp::Any visitWhileStat(LuaParser::WhileStatContext* context) override; antlrcpp::Any visitRepeatStat(LuaParser::RepeatStatContext* context) override;
```
Comment (Koihik) [inline src/FormatVisitor.h]: line break

**Comment by Koihik** [comment]
I prefer this two config name: `keep_simple_block_one_line` & `keep_simple_function_one_line`

**Comment by 5lbBookOfGre** [comment]
Thank you for the response!

> I prefer this two config name: `keep_simple_block_one_line` & `keep_simple_function_one_line`

That won't be backward compatiblity for people who have been using `keep_simple_block_one_line=false`. I don't know how much you care about the backward compatibility, so if you don't care much about it, then please let me know. I will fix it.

**Comment by Koihik** [comment]
> Thank you for the response!
> 
> > I prefer this two config name: `keep_simple_block_one_line` & `keep_simple_function_one_line`
> 
> That won't be backward compatiblity for people who have been using `keep_simple_block_one_line=false`. I don't know how much you care about the backward compatibility, so if you don't care much about it, then please let me know. I will fix it.

I thought backward compatiblity don't have to be that strict.

**Comment by 5lbBookOfGre** [comment]
Updated. Using `keep_simple_control_block_one_line` instead of `keep_simple_block_one_line`, because both function body and control blocks are called as "block" in the code and I do not want to confuse future readers.

Thanks!

**Comment by 5lbBookOfGre** [comment]
Shoot, found a bug in the test file. Will fix it later.

**Pull request merged**